crates/curp-external-api/src/cmd.rs

```diff
@@ -44,7 +44,7 @@ pub trait Command: pri::Serializable + ConflictCheck + PbCodec {
     type ASR: pri::Serializable + PbCodec;
 
     /// Get keys of the command
-    fn keys(&self) -> &[Self::K];
+    fn keys(&self) -> Vec<Self::K>;
```

Comment:
Why do you return a `Vec` instead of `&[]`?

Reply:
The underlying method returned an owned type of the keys
```rust
impl RequestWrapper {
    /// Get keys of the request
    pub fn keys(&self) -> Vec<KeyRange> {
        match *self {
            RequestWrapper::RangeRequest(ref req) => req.keys(),
            RequestWrapper::PutRequest(ref req) => req.keys(),
            RequestWrapper::DeleteRangeRequest(ref req) => req.keys(),
            RequestWrapper::TxnRequest(ref req) => req.keys(),
            _ => vec![],
        }
    }
...
}

Reply:
Shall we implement a `key_ref` method to avoid unnecessary copying?

Reply:
Currently the `KeyRange` requires copying the keys of the command. Implementing a `key_ref` may involves significant change to the `KeyRange` itself.
```rust
impl KeyRange {
    /// New `KeyRange`
    #[inline]
    pub fn new(start: impl Into<Vec<u8>>, end: impl Into<Vec<u8>>) -> Self {
        let key_vec = start.into();
        let range_end_vec = end.into();
        ...
    }
```
I'll leave a TODO to refactor this in the future.